frontend/public/components/factory/details-breadcrumb-resolver.tsx

```diff
@@ -0,0 +1,31 @@
+import * as React from 'react';
+import { match } from 'react-router';
+import { DetailsPageBreadCrumbsHook } from '@console/plugin-sdk';
+import { K8sKind } from '@console/internal/module/k8s';
+
+type DetailsBreadcrumbResolverType = {
+  useBreadcrumbs: DetailsPageBreadCrumbsHook;
+  onBreadcrumbsResolved: (
+    breadcrumbs: {
+      name: string;
+      path: string;
+    }[],
+  ) => void;
+  kind: K8sKind;
+  urlMatch: match<any>;
+};
+
+const DetailsBreadcrumbResolver: React.FC<DetailsBreadcrumbResolverType> = ({
+  useBreadcrumbs,
+  onBreadcrumbsResolved,
+  kind,
+  urlMatch,
+}) => {
+  const breadcrumbs = useBreadcrumbs(kind, urlMatch);
+  React.useEffect(() => {
+    onBreadcrumbsResolved(breadcrumbs);
```

Comment:
There should be a check for value returned by `useBreadcrumbs` hook. An async hook can lead to callback being called with `undefined`.